Solutions/PaloAlto-PAN-OS/Playbooks/PlaybooksOnPrem/PaloAlto-PAN-OS-GetURLCategoryInfo/readme.md

```diff
@@ -0,0 +1,99 @@
+    * CustomConnectorName: Name of the custom REST API connector which interacts with PAN-OS
+    * KeyVaultName: Name of Azure Key Vault that will store X-PAN-KEY
+    * secretName: Name of the secret that will be stored in Key vault
+    * OnPremiseGatewayName: On-premises data gateway that will be used with PaloAlto connector.
+    
+### Post-Deployment instructions 
+#### a. Authorize connections
+Once deployment is complete, you will need to authorize each connection.
+1.	Click the Azure Sentinel connection resource
+2.	Click edit API connection
+3.	Click Authorize
+4.	Sign in
+5.	Click Save
+6.	Repeat steps for other connection such as Teams connection.
+
+#### b. Configurations in Sentinel
+1. In Azure sentinel analytical rules should be configured to trigger an incident with risky user account 
+2. Configure the automation rules to trigger this playbook
+
+
+#### c. Configuration of Azure Key Vault
+1. Navigate to  created Azure Key Vault.
+   ![Key Vault configuration](./images/KeyVault.png)
+2. Create new Access Police with secret Get permission
+   ![Secret permission creation](./images/CreatePolice.png)
+3. Find principal by playbook name and select it
+4. Click Create
+
+[![Deploy to Azure](https://aka.ms/deploytoazurebutton)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fgithub.com%2Fsocprime%2FAzure-Sentinel%2Fraw%2FPAN-OS-OnPremCustomConnector%2FPlaybooks%2FPaloAlto-PAN-OS%2FPlaybooksOnPrem%2FPaloAlto-PAN-OS-GetURLCategoryInfo%2Fazuredeploy.json)  
+[![Deploy to Azure Gov](https://aka.ms/deploytoazuregovbutton)](https://portal.azure.us/#create/Microsoft.Template/uri/https%3A%2F%2Fgithub.com%2Fsocprime%2FAzure-Sentinel%2Fraw%2FPAN-OS-OnPremCustomConnector%2FPlaybooks%2FPaloAlto-PAN-OS%2FPlaybooksOnPrem%2FPaloAlto-PAN-OS-GetURLCategoryInfo%2Fazuredeploy.json)
+
+## Playbook steps explained
+
+### When Azure Sentinel incident creation rule is triggered
+
+Azure Sentinel incident is created. The playbook receives the incident as the input.
+
+### Get Secret
+Gets X-PAN-KEY from created Azure Key Vault
+
+### Entities - Get URLs
+
+Get the list of risky/malicious URLs as entities from the Incident
+
+### List-address objects
+
+Playbook uses "List address objects" action to get address object details from PAN-OS
+
+### List URL filtering category information
+
+Playbook uses "List URL filtering category information" action to get URL filtering category details from PAN-OS
+### For each-risky URL received from the incident
+Iterates on the URLs found in this incident (probably one) and performs the following:
+
+1. For the risky URL, Filter URL from list of address objects action to get specific address object details from PAN-OS
+
+   a. Compose body of address object where URL is a member for updating incident with address object details
+
+ 2. Create HTML table for URL category information such as name, location and description
+
+ 3. Add a comment to the incident with the information below:
```

Comment:
Any playbook, which requires updating incident with comments, that playbook should be having "Microsoft Sentinel Responder Role" assigned, else comment action of logic app will generate an authentication error

For reference, please take help from below readme file:
https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/ThreatXCloud/Playbooks/ThreatXPlaybooks/ThreatX-BlockIP-URL

Reply:
Kindly check all playbooks for the same , because all playbooks which has step or action to add comment from logic app to sentinel incident would be needing this role .